src/coremidi.cs

```diff
@@ -228,5 +253,172 @@ interface MidiCISession
+	[DisableDefaultCtor]
+	interface MidiCIDeviceInfo : NSSecureCoding
+	{
+		[Export ("manufacturerID")]
+		NSData ManufacturerID { get; }
+
+		[Export ("family")]
+		NSData Family { get; }
+
+		[Export ("modelNumber")]
+		NSData ModelNumber { get; }
+
+		[Export ("revisionLevel")]
+		NSData RevisionLevel { get; }
+
+		[Internal]
+		[Export ("midiDestination")]
+		MidiObjectRef _MidiDestination { get; }
+
+		MidiEndpoint MidiDestination {
+			[Wrap ("new MidiEndpoint (_MidiDestination)")]
+			get;
+		}
+
+		[Internal]
+		[Export ("initWithDestination:manufacturer:family:model:revision:")]
+		IntPtr Constructor (MidiObjectRef midiDestination, NSData manufacturer, NSData family, NSData modelNumber, NSData revisionLevel);
+
+		[Wrap ("this (midiDestination.Handle, manufacturer, family, modelNumber, revisionLevel)")]
+		IntPtr Constructor (MidiEndpoint midiDestination, NSData manufacturer, NSData family, NSData modelNumber, NSData revisionLevel);
+	}
+
+	[Mac (10, 16), iOS (14, 0)]
+	[BaseType (typeof (NSObject), Name="MIDICIDiscoveredNode")]
+	[DisableDefaultCtor]
+	interface MidiCIDiscoveredNode
+	{
+		[Internal]
+		[Export ("destination")]
+		MidiObjectRef _Destination { get; }
+
+		MidiEndpoint Destination { 
+			[Wrap ("new MidiEndpoint (_Destination)")]
+			get;
+		}
+
+		[Export ("deviceInfo")]
+		MidiCIDeviceInfo DeviceInfo { get; }
+
+		[Export ("supportsProfiles")]
+		bool SupportsProfiles { get; }
+
+		[Export ("supportsProperties")]
+		bool SupportsProperties { get; }
+
+		[Export ("maximumSysExSize")]
+		NSNumber MaximumSysExSize { get; }
+	}
+
+	delegate void MidiCIDiscoveryResponseDelegate (MidiCIDiscoveredNode[] arg0);
+
+	[Mac (10, 16), iOS (14, 0)]
+	[BaseType (typeof (NSObject), Name="MIDICIDiscoveryManager")]
+	interface MidiCIDiscoveryManager
+	{
+		[Static]
+		[Export ("sharedInstance")]
+		MidiCIDiscoveryManager SharedInstance { get; }
+
+		[Export ("discoverWithHandler:")]
+		void Discover (MidiCIDiscoveryResponseDelegate completedHandler);
+	}
+
+	interface IMidiCIProfileResponderDelegate {} 
+
+	[Mac (10,16), iOS (14,0)]
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject), Name="MIDICIProfileResponderDelegate")]
+	interface MidiCIProfileResponderDelegate
+	{
+		[Abstract]
+		[Export ("connectInitiator:withDeviceInfo:")]
+		bool ConnectInitiator (NSNumber initiatorMuid, MidiCIDeviceInfo deviceInfo);
+
+		[Abstract]
+		[Export ("initiatorDisconnected:")]
+		void InitiatorDisconnected (NSNumber initiatorMuid);
+
+		[Export ("willSetProfile:onChannel:enabled:")]
+		bool WillSetProfile (MidiCIProfile profile, byte channel, bool shouldEnable);
+
+		[Export ("handleDataForProfile:onChannel:data:")]
+		void HandleData (MidiCIProfile profile, byte channel, NSData inData);
+	}
+
+	[Mac (10, 16), iOS (14, 0)]
+	[BaseType (typeof (NSObject), Name="MIDICIResponder")]
+	[DisableDefaultCtor]
+	interface MidiCIResponder
+	{
+		// MIDICIInitiatiorMUID is a type alias to NSNumber, we do not have much more info
```

Comment:
seems to be 32 bits
https://www.midi.org/forum/5674-question-after-reading-the-midi-ci-1-1-spec-re-muids
or even 28 bits is you read further :)

Reply:
heh nice find